massa-protocol-worker-2/src/handlers/peer_handler/tester.rs

```diff
@@ -64,41 +65,67 @@ impl HandshakeHandler for TesterHandshake {
     ) -> PeerNetResult<PeerId> {
         let data = endpoint.receive()?;
         let peer_id = PeerId::from_bytes(&data[..32].try_into().unwrap())?;
-        let (_, announcement) = self
-            .announcement_deserializer
-            .deserialize::<DeserializeError>(&data[32..])
-            .map_err(|err| {
-                PeerNetError::HandshakeError.error(
-                    "Tester Handshake",
-                    Some(format!("Failed to deserialize announcement: {}", err)),
-                )
-            })?;
-        if peer_id
-            .verify_signature(&announcement.hash, &announcement.signature)
-            .is_err()
-        {
-            return Err(PeerNetError::HandshakeError
-                .error("Tester Handshake", Some(String::from("Invalid signature"))));
-        }
-        //TODO: Check ip we are connected match one of the announced ips
-        {
+
+        let res = {
+            {
+                let mut peer_db_write = self.peer_db.write();
+                peer_db_write
+                    .peers
+                    .entry(peer_id.clone())
+                    .and_modify(|info| {
+                        info.state = super::PeerState::InHandshake;
+                    });
+            }
+
+            let (_, announcement) = self
+                .announcement_deserializer
+                .deserialize::<DeserializeError>(&data[32..])
+                .map_err(|err| {
+                    PeerNetError::HandshakeError.error(
+                        "Tester Handshake",
+                        Some(format!("Failed to deserialize announcement: {}", err)),
+                    )
+                })?;
+            if peer_id
+                .verify_signature(&announcement.hash, &announcement.signature)
+                .is_err()
+            {
+                return Err(PeerNetError::HandshakeError
+                    .error("Tester Handshake", Some(String::from("Invalid signature"))));
+            }
+            //TODO: Check ip we are connected match one of the announced ips
+            {
+                let mut peer_db_write = self.peer_db.write();
+                peer_db_write
+                    .index_by_newest
+                    .insert(announcement.timestamp, peer_id.clone());
+                peer_db_write
+                    .peers
+                    .entry(peer_id.clone())
+                    .and_modify(|info| {
+                        if info.last_announce.timestamp < announcement.timestamp {
+                            info.last_announce = announcement.clone();
+                        }
+                        info.state = super::PeerState::HandshakeSuccess;
```

Comment:
The `HandshakeSuccess` is strange to me. In all cases it shouldn't be trigger in the handshake of the Tester. The tester is here to determine only if the listeners we received are indeed receivers that we can connect to to join the massa network and so the peer is trusted.

Also be careful if someone is already banned we need to filter them before trying to test them and we need to don't accept their connections